assert/assertions.go

```diff
@@ -1948,19 +1948,28 @@ func diff(expected interface{}, actual interface{}) string {
 
 	var e, a string
 
-	switch et {
-	case reflect.TypeOf(""):
-		e = reflect.ValueOf(expected).String()
-		a = reflect.ValueOf(actual).String()
-	case reflect.TypeOf(time.Time{}):
-		e = spewConfigStringerEnabled.Sdump(expected)
-		a = spewConfigStringerEnabled.Sdump(actual)
-	default:
-		e = spewConfig.Sdump(expected)
-		a = spewConfig.Sdump(actual)
+	// Use spew to create string representations of the objects
+	// We have to guard against panics in spew.
+	// See https://github.com/stretchr/testify/issues/480
+	panicked, _, _ := didPanic(func() {
+		switch et {
+		case reflect.TypeOf(""):
+			e = reflect.ValueOf(expected).String()
+			a = reflect.ValueOf(actual).String()
+		case reflect.TypeOf(time.Time{}):
+			e = spewConfigStringerEnabled.Sdump(expected)
+			a = spewConfigStringerEnabled.Sdump(actual)
+		default:
+			e = spewConfig.Sdump(expected)
+			a = spewConfig.Sdump(actual)
+		}
+	})
+	if panicked {
```

Comment:
Panics are not exceptions, we should not be blindly ignoring them. If there is a case we know of that spew can't handle then we should avoid it or fix spew, potentially by vendoring it.